### ISSUE TITLE
UWP App -> System.DllNotFoundException: Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

### ISSUE THREAD

**Issue by ataparia** [praeclarum/sqlite-net#814]
I am getting the following error while trying to run UWP app that has reference to "sqlite-net-pcl".
My situation is little complicated. My project has this file reference "SQLiteNet.cs" which is the source code for SQLite interaction written by Krueger Systems. I had to manipulate this file to adjust to my projects demands. 

The error happens in the constructor of SQLiteConnection class in this particular line:
SQLite3.SetDirectory(/*temp directory type*/2, Windows.Storage.ApplicationData.Current.TemporaryFolder.Path);

Any help to fix this issue will be greatly appreciated.

Thanks
Amit

**Comment by ericsink** [comment]
I assume that your SQLiteNet.cs is a modified copy of SQLite.cs from here in this repo.

So the first thing to say is that if you are using this library in source code form, then you don't need sqlite-net-pcl.  People use this library either through the nuget package or the source code.  One or the other, not both.  :-)

**Comment by ataparia** [comment]
Thanks for the quick response Eric.

I got the SQLite source code from here - https://github.com/praeclarum/sqlite-net/blob/master/src/SQLite.cs 

I removed the nuget  "sqlite-net-pcl" but still continue to get the same error.

I think I would need some other reference(may be some other dll) as the static SQLite3 class does have mapping to some SQLite3 library. Something like this:

const string LibraryPath = "sqlite3";
[DllImport(LibraryPath, EntryPoint = "sqlite3_open", CallingConvention = CallingConvention.Cdecl)] public static extern Result Open([MarshalAs(UnmanagedType.LPStr)] string filename, out IntPtr db);
